scripts/common.sh

```diff
@@ -26,18 +26,18 @@ set -e
 # changes directory. The paths returned by `caller` are relative to it.
 _initial_work_dir=$(pwd)
 
-if [ "$CIRCLECI" ]
+if [[ ${CIRCLECI:-} != "" ]]
```

Comment:
Why does the test need to check if it differs from an empty string explicitly? To test if some environment variable is not defined, I usually use `-n` that checks whether the length of a string is non-zero (I think this is the default behavior, i.e., `[[ "$CIRCLECI" ]]`). And as you are substituting it with nothing, it will return an empty string if `CIRCLECI` is not defined. So if the idea was to be more explicit about the test, you could just add the `-n`:
```suggestion
if [[ -n "$CIRCLECI" ]]
```

Reply:
The problem is that this script might be included in another script where the check for unbound variables via `set -u` was enabled (as with the `scripts/ASTImportTest.sh` script). That means `$CIRCLECI` does only have a real value, if the corresponding environment variables was defined. The script fails otherwise with something like `scripts/common.sh: line 29: CIRCLECI: unbound variable`. So doing this strange substitution ensures that `$CIRCLECI` is always defined, even if executed outside circleci. It will just be empty in that case.

Reply:
Ah, right. So it is fine to keep in that way since the variable must to be set:)